ecs-init/cache/cache_test.go

```diff
@@ -122,13 +123,18 @@ func TestDownloadAgentDownloadMD5Failure(t *testing.T) {
 
 	mockFS := NewMockfileSystem(mockCtrl)
 	mockGetter := NewMockhttpGetter(mockCtrl)
+	mockMetadata := NewMockinstanceMetadata(mockCtrl)
 
-	mockFS.EXPECT().MkdirAll(config.CacheDirectory(), os.ModeDir|0700)
-	mockGetter.EXPECT().Get(config.AgentRemoteTarballMD5()).Return(nil, errors.New("test error"))
+	gomock.InOrder(
+		mockFS.EXPECT().MkdirAll(config.CacheDirectory(), os.ModeDir|0700),
+		mockMetadata.EXPECT().Region().Return(config.DefaultRegionName, nil),
```

Comment:
Unforunately with only two regions, the default case is the normal path. Currently this works as an exception list, only including custom values for special regions that override the default. As additional regions are added these tests should be added. For now, this ambiguity is the result of attempting to de-duplicate a previous change that included specifying another region as well as a default.

The negative test case has been added.